src/dict.c

```diff
@@ -640,7 +640,7 @@ int dictReplace(dict *d, void *key, void *val) {
      * reverse. */
     void *oldval = dictGetVal(existing);
     dictSetVal(d, existing, val);
-    if (d->type->valDestructor) d->type->valDestructor(d, oldval);
+    if (d->type->valDestructor) d->type->valDestructor(oldval);
```

Comment:
Here and in some other places, the callbacks are already called directly without using the helper functions. We could do that everywhere actually. No need for dictFreeVal.

Reply:
This is the only place where it is directly invoking `valDestructor()` which is for the case when the old value needs to be freed. Also from a high level code API POV, this code along with `dictFreeVal()`, are purely internal implementations inside `dict.c` and has no effect to the API definitions that are exposed to the outside. Don't think this is a big issue.